bundle/manifests/self-node-remediation.clusterserviceversion.yaml

```diff
@@ -337,6 +337,16 @@ spec:
                 control-plane: controller-manager
                 self-node-remediation-operator: ""
             spec:
+              affinity:
+                podAntiAffinity:
+                  requiredDuringSchedulingIgnoredDuringExecution:
```

Comment:
> Would this fail operator installation in case we only have one node available during installation?

I will test it this case. In my opinion, I think it should fail to install the operator installation.
There are the following cases:
1. We have only one node available like SNO(Single Node OpenShift)
    => No reason to deploy SNR/FAR/NHC
2. We have a plan to deploy a cluster with multiple nodes, but there is only one node available during installation due to the node failures
    => We are unhappy to succeed to deploy the cluster without noticing the node failures. It should stop the operator installation.



> looks like the update concern can also be solved with this:
> 
> ```
> strategy:
>     rollingUpdate:
>       maxSurge: 0
>       maxUnavailable: 1
>     type: RollingUpdate
> ```
> 
> it will force k8s to first delete one of the pods before starting a new one

I will fix it with the above. And the Workload Availability project has the Node Maintenance Operator for maintenance. So we also need to define PodDisruptionBudget like https://github.com/openshift/cluster-monitoring-operator/blob/master/assets/thanos-querier/pod-disruption-budget.yaml.
It will prevent from stopping all of the managers accidentally.

Reply:
> We have only one node available like SNO(Single Node OpenShift)
> => No reason to deploy SNR/FAR/NHC

agree

> We have a plan to deploy a cluster with multiple nodes, but there is only one node available during installation due to the node failures
> => We are unhappy to succeed to deploy the cluster without noticing the node failures. It should stop the operator installation.

hm.... yes, actually it does not make sense to succeed, it wouldn't help fixing the cluster, because the agents would not be deployed on the failed nodes.

> we also need to define PodDisruptionBudget

Let's discuss this in an issue please. As with this affinity, we need to be aware of side effects. Let's not make our operators more important than they are, compared to other deployments.